include/deal.II/lac/block_linear_operator.h

```diff
@@ -351,6 +351,43 @@ namespace internal
 {
   namespace BlockLinearOperatorImplementation
```

Comment:
Can you put these two helper functions in an anonymous namespace please, i.e.
```
namespace 
  {
    // helper functions...
  }
```

We have `internal::LinearOperatorImplementation` and `internal::BlockLinearOperatorImplementation` for things that need to be overriden in other header files (reinit, payload). The corresponding `apply_with_intermediate_storage` etc. is in an anonymous namespace.